datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -540,38 +540,6 @@ impl LogicalPlan {
         }
     }
 
-    /// Returns a copy of this `LogicalPlan` with the new inputs
-    pub fn with_new_inputs(&self, inputs: &[LogicalPlan]) -> Result<LogicalPlan> {
```

Comment:
I suggest we mark this deprecated for at least a release or two to help users migrate? I always apprecate the hint in the deprecation message that tells me what to change so I don't have to go poking around in the commit log to see what happened.

 Perhaps it could call through to the `with_new_exprs` API, something like (untested):

```rust
    #[deprecated(since = "35.0.0", note = "please use `with_new_exprs` instead")]
    pub fn with_new_inputs(&self, inputs: &[LogicalPlan]) -> Result<LogicalPlan> {
      self.with_new_exprs(self, self.expressions(), inputs)
    }
```

Reply:
Agreed. Marked it as deprecated one.